Source/WebKit/UIProcess/SwiftDemoLogo.swift

```diff
@@ -0,0 +1,164 @@
+// Copyright (C) 2025 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+
+#if ENABLE_SWIFT_DEMO_URI_SCHEME
+
+import Foundation
+internal import WebKit_Internal_Cxx
+import struct Swift.String
+
+extension Data {
+    var bytes: [UInt8] {
+        [UInt8](self)
+    }
+}
+
+// This function is contrived to exercise bi-directional C++ -> Swift -> C++
+// calls within the main WebKit target. It can be tested by enabling
+// ENABLE_SWIFT_DEMO_DATA_URL_ENCODING, and then visiting
+// x-swift-demo:// in the browser. The goal is to provide an easy
+// but comprehensive test case to determine whether Swift-C++ interop
+// is working in a given toolchain.
+// We make this function "available" to C++ on all SDK versions, but it's
+// not truly available to clients since we do not export it from the dylib.
+/// Return data for a PNG of a Swift logo.
+@available(iOS 15.0, macOS 15.0, tvOS 15.0, watchOS 9.0, *)
+@_expose(Cxx)
+public func getSwiftLogoData() -> [UInt8] {
```

Comment:
Out of curiosity, does this function end up in WebKit's public swiftmodule when this feature is turned on? That seems like a problem for real adoption.

Reply:
Presumably it's in the textual .swiftinterface too, I think that's what actually matters. Yes, this shouldn't block this PR.